packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -1,12 +1,29 @@
 import { NotifiService } from '@notifi-network/notifi-graphql';
-import { UserFragmentFragment } from 'notifi-graphql/lib/gql/generated';
 
+import {
+  BeginLogInByTransactionResult,
+  CompleteLogInByTransactionInput,
+  CompleteLogInByTransactionMutation,
+  UserFragmentFragment,
+  WalletBlockchain,
+} from '../../notifi-graphql/lib/gql/generated';
 import type { NotifiFrontendConfiguration } from './configuration/NotifiFrontendConfiguration';
 import type { NotifiStorage } from './storage';
 import { notNullOrEmpty } from './utils/notNullOrEmpty';
 
 export type SignMessageFunction = (message: Uint8Array) => Promise<Uint8Array>;
 
+type BeginLoginProps = {
+  walletBlockchain: WalletBlockchain;
+  walletPublicKey: string;
+};
+
+type CompleteLoginProps = {
+  input: CompleteLogInByTransactionInput;
+};
+
+let clientRandomUuid: string;
```

Comment:
`let clientRandomUuid: string | null = null`

Also, this should be an instance variable, not a module scope variable